developer-guide/development/controller-mamager-quickstart.md

```diff
@@ -0,0 +1,145 @@
+# Controller Manager Quickstart
+
+KubeSphere follows the [sample-controller](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-controller) pattern to implement controllers for the reconciliation process. It watches CustomResourceDefinition(CRD) desired state and actual state through informers and listers. Then, it sends instructions through clients to try and make the current state be more like the desired state.
+
+> We won't discuss the pros and cons of the sample-controller comparing with Kubebuilder here. Just show you how to implement a controller.
+
+**Note:** Read `sample-controller`'s [README.md](https://github.com/kubernetes/sample-controller/blob/master/README.md) and go through the `controller.go` before you begin.
+
+## Defining types
+
+To implement a controller, you should define your API Groups, Versions, and Kinds firstly. Please read more about the [Kubernetes API Conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#websockets-and-spdy) if you want to understand those concepts deeper.
+
+For an example, the `Group` Kind that was defined under `iam.kubershpere.io` API Group as below:
+
+```golang
+// +genclient:nonNamespaced
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// +kubebuilder:printcolumn:name="Workspace",type="string",JSONPath=".metadata.labels.kubesphere\\.io/workspace"
+// +kubebuilder:resource:categories="group",scope="Cluster"
+
+// Group is the Schema for the groups API
+type Group struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+    Status GroupStatus `json:"status,omitempty"`
```

Comment:
I guess you just missed here. Please provide a good indent code for all code lines.